doc/api/next_api_changes/deprecations/23668-TC.rst

```diff
@@ -0,0 +1,14 @@
+Pending deprecation top-level cmap registration and access functions in ``mpl.cm``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As part of a `multi-step process
+<https://github.com/matplotlib/matplotlib/issues/20853>`_ we are refactoring
+the global state for managing the registered colormaps.
+
+In Matplotlib 3.5 we added a `.ColormapRegistry` class and exposed an
+instance at the top level as ``matplotlib.colormaps``.  The existing
+top level functions in `matplotlib.cm` (``get_cmap``, ``register_cmap``,
```

Comment:
I just realized that `get_cmap` and `register_cmap` are imported into `pyplot`, which makes the deprecation active also there.

https://github.com/matplotlib/matplotlib/blob/2b96795e324db95509ed1f669c471919685056d6/lib/matplotlib/pyplot.py#L71

IMHO it's ok to deprecate `register_cmap` in `pyplot`. However, I think that `plt.get_cmap()` is too widely used to be deprecated. Therefore we should implement a `plt.get_cmap()` adapter in `pyplot` (maybe deprecate the `lut` parameter there in the future). If that's too much for 3.6, it's ok to only add the pending deprecation to `register_cmap` and `unregister_cmap` and handle `get_cmap` in 3.7.

Reply:
So the tasks here are:

 - implement a non-warning `get_cmap` in pyplot
 - track down everyplace `register_cmap` and `unregister_cmap` are imported and make sure their deprecation is noted
 - update deprecation note to note all of the above

Reply:
I believe this is done.